include/dlaf/matrix/matrix_base.h

```diff
@@ -46,49 +46,77 @@ class MatrixBase {
     return distribution_.size();
   }
 
-  /// Returns the block size of the matrix.
+  /// Returns the complete block size of the matrix.
+  const TileElementSize& block_size() const noexcept {
+    return distribution_.block_size();
+  }
+
+  /// Returns the complete tile size of the matrix.
+  const TileElementSize& tile_size() const noexcept {
+    return distribution_.tile_size();
+  }
+
+  /// Returns the number of tiles of the global matrix (2D size).
+  const GlobalTileSize& nr_tiles() const noexcept {
+    return distribution_.nr_tiles();
+  }
+
+  /// Returns the id associated to the matrix of this rank.
+  const comm::Index2D& rank_index() const noexcept {
+    return distribution_.rank_index();
+  }
+
+  /// Returns the size of the communicator grid associated to the matrix.
+  const comm::Size2D& grid_size() const noexcept {
+    return distribution_.grid_size();
+  }
+
+  /// Returns the 2D rank index of the process that stores the tile with global index @p global_tile.
+  ///
+  /// @pre global_tile.isIn(nrTiles()).
+  comm::Index2D rank_global_tile(const GlobalTileIndex& global_tile) const noexcept {
+    return distribution_.rank_global_tile(global_tile);
+  }
+
+  /// Returns the size of the Tile with global index @p index.
+  TileElementSize tile_size_of(const GlobalTileIndex& index) const noexcept {
+    return distribution_.tile_size_of(index);
+  }
+
+  /// Returns the distribution of the matrix.
+  const matrix::Distribution& distribution() const noexcept {
+    return distribution_;
+  }
+
+  // TODO remove deprecated
```

Comment:
Also TODO. Are you planning to remove the deprecated functions in this PR, or just mark them deprecated for now?